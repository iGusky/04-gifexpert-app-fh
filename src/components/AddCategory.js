import React from 'react'
import { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ( {setCategorias} ) => {

  const [inputValue, setInputValue ] = useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  }
 
  const handleSubmit = (e) =>{
    e.preventDefault();
    if(inputValue.trim().length>2){
      setCategorias( cats => [ inputValue, ...cats]);
      setInputValue('');
    }
  }

  return (
    <form onSubmit={ handleSubmit } >
        <p>{inputValue}</p>
      <input
        type = "text"
        value = {inputValue}
        onChange = {(e) => handleInputChange(e)}
      />
    </form>
  )
}

AddCategory.propTypes = {
  setCategorias: PropTypes.func.isRequired
};
