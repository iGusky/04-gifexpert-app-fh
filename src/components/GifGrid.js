import React from 'react'
// import {useState, useEffect} from 'react';
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGrid = ({ category }) => {

    //*Antes de usar Customhooks
    // const [images, setImages] = useState([]);
    //Con este codigo hacemos que el getGifs se ejecute solo cuando se rendereiza por primera vez.
    // useEffect( () => {
    //     getGifs( category )
    //         .then( imgs => setImages(imgs));
    // }, [ category ]); //Si la categoria cambia, vuelve a ejecutar el efecto interno.

    //* Usando Customhooks
    //Se renombra data a images
    const {data:images, loading } = useFetchGifs( category );

    return (
    <>
        <h3 className="animate__animated animate__fadeIn">{ category }</h3>

        {/* Forma corta de un if */}
        { loading && <p className="animate__animated animate__flash">Cargando...</p>}

        <div className="card-grid">
            {
                images.map((img) => {
                    return <GifGridItem
                        key = {img.id}
                        {...img}
                    />
                })
            }
        </div>
    </>
    )
}