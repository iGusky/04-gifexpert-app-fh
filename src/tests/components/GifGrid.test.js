import '@testing-library/jest-dom';
import { shallow } from 'enzyme'
import { GifGrid } from '../../components/GifGrid'
import { useFetchGifs } from '../../hooks/useFetchGifs'
jest.mock('../../hooks/useFetchGifs');

describe('Pruebas del componente <GifGrid />',() => {
    
    const category = 'evangelion';

   
    
    test('Debe hacer match con el snapshot', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });

        const wrapper = shallow(<GifGrid category = { category } />);
        expect( wrapper ).toMatchSnapshot();
    });

    test('Debe mostrar items cuando se cargan imagenes', () =>{

        const gifs = [{
            id: 'ABC',
            url: 'https://localhost/img.png',
            title: 'Cualquier titulo'
        },
        {
            id: 'DFG',
            url: 'https://localhost/img.png',
            title: 'Cualquier titulo'
        }];
        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        const wrapper = shallow(<GifGrid category = { category } />);
        expect ( wrapper.find('p').exists() ).toBe(false);
        expect ( wrapper.find('GifGridItem').length ).toBe( gifs.length );

    });
    
})