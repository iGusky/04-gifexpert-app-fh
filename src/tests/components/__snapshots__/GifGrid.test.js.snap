// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas del componente <GifGrid /> Debe hacer match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    evangelion
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Cargando...
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
